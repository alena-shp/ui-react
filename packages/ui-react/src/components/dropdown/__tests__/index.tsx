import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';

import { Dropdown } from '..';
import { IconArrowsChevronRight } from '../../icons/arrows/chevron-right';
import { IconDownload } from '../../icons/download';

const dropdownOptions = [{
    label: 'Option 1',
    value: '1',
    handleClick: () => console.info('Click option 1')
}, {
    label: 'Option 2',
    value: '2',
    handleClick: () => console.info('Click option 2')
}, {
    label: 'Option 3',
    value: '3',
    handleClick: () => console.info('Click option 3')
}];

test('Default use', async () => {
    const { asFragment, getByText } = render(
        <Dropdown
            triggerElement={{
                elRightIcon: 'dots',
                triggerText: 'Trigger'
            }}
            render={(isOpen, onClose) => {
                return dropdownOptions.map((option, index) => (
                    <div
                        key={index}
                        onClick={() => {
                            option.handleClick();

                            onClose();
                        }}
                    >
                        {option.label}
                    </div>
                ));
            }}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Default use icon dots', async () => {
    const { asFragment, getByText } = render(
        <Dropdown
            triggerElement={{
                elRightIcon: 'chevron',
                triggerText: 'Trigger'
            }}
            render={(isOpen, onClose) => {
                return dropdownOptions.map((option, index) => (
                    <div
                        key={index}
                        onClick={() => {
                            option.handleClick();

                            onClose();
                        }}
                    >
                        {option.label}
                    </div>
                ));
            }}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Default use', async () => {
    const { asFragment, getByText } = render(
        <Dropdown
            triggerElement={{
                elRightIcon: <IconArrowsChevronRight />,
                elLeftIcon: <IconDownload />,
                triggerText: 'Trigger'
            }}
            render={(isOpen, onClose) => {
                return dropdownOptions.map((option, index) => (
                    <div
                        key={index}
                        onClick={() => {
                            option.handleClick();

                            onClose();
                        }}
                    >
                        {option.label}
                    </div>
                ));
            }}
        />
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

