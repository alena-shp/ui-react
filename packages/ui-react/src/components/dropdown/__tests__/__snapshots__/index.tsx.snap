// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default use 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <div
        class="dropdown__trigger"
      >
        Trigger
        <svg
          class="dropdown__trigger-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10 5a2 2 0 1 0 4 0 2 2 0 0 0-4 0Zm2 9a2 2 0 1 1 0-4 2 2 0 0 1 0 4Zm0 7a2 2 0 1 1 0-4 2 2 0 0 1 0 4Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default use 2`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <div
        class="dropdown__trigger"
      >
        Trigger
        <svg
          class="dropdown__trigger-icon dropdown__trigger-icon_active-dots"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10 5a2 2 0 1 0 4 0 2 2 0 0 0-4 0Zm2 9a2 2 0 1 1 0-4 2 2 0 0 1 0 4Zm0 7a2 2 0 1 1 0-4 2 2 0 0 1 0 4Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </span>
    <span
      class="dropdown__content"
      style="position: fixed; top: 0px; left: 0px; min-width: 0;"
    >
      <div>
        Option 1
      </div>
      <div>
        Option 2
      </div>
      <div>
        Option 3
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default use icon dots 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <div
        class="dropdown__trigger"
      >
        Trigger
        <svg
          class="dropdown__trigger-icon dropdown__trigger-icon_chevron"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="m19.293 7.293 1.414 1.414L12 17.414 3.293 8.707l1.414-1.414L12 14.586l7.293-7.293Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Default use icon dots 2`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <div
        class="dropdown__trigger"
      >
        Trigger
        <svg
          class="dropdown__trigger-icon dropdown__trigger-icon_chevron dropdown__trigger-icon_active-chevron"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="m19.293 7.293 1.414 1.414L12 17.414 3.293 8.707l1.414-1.414L12 14.586l7.293-7.293Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </span>
    <span
      class="dropdown__content"
      style="position: fixed; top: 0px; left: 0px; min-width: 0;"
    >
      <div>
        Option 1
      </div>
      <div>
        Option 2
      </div>
      <div>
        Option 3
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Trigger button 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="button button_medium button_default"
        type="button"
      >
        Button
      </button>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Trigger button 2`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <button
        class="button button_medium button_default"
        type="button"
      >
        Button
      </button>
    </span>
    <span
      class="dropdown__content"
      style="position: fixed; top: 0px; left: 0px; min-width: 0;"
    >
      <div>
        Option 1
      </div>
      <div>
        Option 2
      </div>
      <div>
        Option 3
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Trigger custom icon 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <svg
        height="24"
        role="Trigger"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M14.586 12 7.293 4.707l1.414-1.414L17.414 12l-8.707 8.707-1.414-1.414L14.586 12Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Trigger custom icon 2`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <svg
        height="24"
        role="Trigger"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M14.586 12 7.293 4.707l1.414-1.414L17.414 12l-8.707 8.707-1.414-1.414L14.586 12Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <span
      class="dropdown__content"
      style="position: fixed; top: 0px; left: 0px; min-width: 0;"
    >
      <div>
        Option 1
      </div>
      <div>
        Option 2
      </div>
      <div>
        Option 3
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Trigger left right icons 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <div
        class="dropdown__trigger"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16.293 9.293 13 12.586V2h-2v10.586L7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707-1.414-1.414ZM22 20v-4h-2v4H4v-4H2v4a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2Z"
            fill-rule="evenodd"
          />
        </svg>
        Trigger
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M14.586 12 7.293 4.707l1.414-1.414L17.414 12l-8.707 8.707-1.414-1.414L14.586 12Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Trigger left right icons 2`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      class="popover__trigger"
    >
      <div
        class="dropdown__trigger"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16.293 9.293 13 12.586V2h-2v10.586L7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707-1.414-1.414ZM22 20v-4h-2v4H4v-4H2v4a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2Z"
            fill-rule="evenodd"
          />
        </svg>
        Trigger
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M14.586 12 7.293 4.707l1.414-1.414L17.414 12l-8.707 8.707-1.414-1.414L14.586 12Z"
            fill-rule="evenodd"
          />
        </svg>
      </div>
    </span>
    <span
      class="dropdown__content"
      style="position: fixed; top: 0px; left: 0px; min-width: 0;"
    >
      <div>
        Option 1
      </div>
      <div>
        Option 2
      </div>
      <div>
        Option 3
      </div>
    </span>
  </div>
</DocumentFragment>
`;
